### ISSUE TITLE
backintime --decode fails

### ISSUE THREAD

**Issue by Germar** [bit-team/backintime#131]
Hi,

while taking a large backup I got a few errors:

### 'backintime --quiet --profile sshencqnap --decode <path>'

========== Take snapshot (profile 4): Wed Aug 27 09:48:12 2014 ==========

[E] Error: rsync: opendir "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/52WwAbIaFKiD0y6G-YepJxlB/Zt4UnmHw4B1ge1vcoAdgWGTa/ikyyPZzmTvYnQdCY7I67GTUIP-W451ieR-oy-1v2qNg5Y-" failed: Permission denied (13)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/,lowW2gfrcA4kmgrqLJQ0HZb/k,QEQ5ytQ1BKIVNtK2lm06-4tLqIw3nNjHzSZyyml5QLDtAd9Bu,FP0jHLgI,OYi688" failed: No such file or directory (2)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/MULqQ08GNdNXVswml,yl8CGn6jceUbK8I,OjouTTDuCRi-/0oznTc8pnjJcnAaXMFjE5rxL" failed: No such file or directory (2)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/fz0iidkcecTJkfnxS0MmJCoi/FykOJyYSwCb4MsSx-SCmc9BK/ljtrAIldBoOSu69PrkIRTUVY/LqndDR7MiKB3GGQF4EaFB3IbX-X-kUBPnD1llyrY-GyC9-" failed: No such file or directory (2)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/ilPZ4b,6a6nZfmaoue13jbsQ/Ij1sNb96P1D3nxmRdyMHJWGn/-XkJor4PI8G2qHzw6yTe,caR" failed: No such file or directory (2)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/ilPZ4b,6a6nZfmaoue13jbsQ/Ij1sNb96P1D3nxmRdyMHJWGn/ZZ-vTZmODusmOJPGth9CGIMl" failed: No such file or directory (2)
[E] Error: rsync: stat "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/ocL0N7IAqZ8DFY5L-Kolwcz0/JvqJax1V1Ht9SsrYYoqi6dF9/DkRNxnJEJxSFJL,4AKRTYa3U/vDQ8EIhIyweoIeVaUW0yeQhZ/160JQ,rAf5gxNrXCksJrvfSU" failed: No such file or directory (2)
[E] Error: rsync: mknod "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/ocL0N7IAqZ8DFY5L-Kolwcz0/JvqJax1V1Ht9SsrYYoqi6dF9/DkRNxnJEJxSFJL,4AKRTYa3U/vDQ8EIhIyweoIeVaUW0yeQhZ/OPSdkEt6OGJf4ipr7OYmCW5V" failed: Address already in use (98)

In fact none of the filenames could be automatically decoded. This is also the case when trying to manually decode:

# backintime --profile sshencqnap --decode "/share/backup/backintime/UqUO2ytveGGjBq3qzGsgX6Py/F,VDzqn3,cAbylQZJNwRWWYT/yqzTd8DssQsLmqZ5pqMeXtiC/0,m,z4Xv,uVJMJMq,7ElnUsh/dvP0RHayype,rqHv0XXDYFfF/grPaOkZAp08oDalSqosv0rZ0/JA-tGYW9HxC21cPbI7smjare/eWn4krRY0gwkUKYWeFJGX9u5/,lowW2gfrcA4kmgrqLJQ0HZb/k,QEQ5ytQ1BKIVNtK2lm06-4tLqIw3nNjHzSZyyml5QLDtAd9Bu,FP0jHLgI,OYi688"

Back In Time
Version: 1.0.34

Back In Time comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type `backintime --license' for details.

INFO: Mountpoint /tmp/backintime/root/mnt/6A6267D7/mountpoint is already mounted
INFO: Mountpoint /tmp/backintime/root/mnt/22177331/mountpoint is already mounted
INFO: Mountpoint /tmp/backintime/root/mnt/B24722EF/mountpoint is already mounted
Traceback (most recent call last):
  File "/usr/share/backintime/common/backintime.py", line 405, in <module>
    start_app()
  File "/usr/share/backintime/common/backintime.py", line 351, in start_app
    ret = decode.list(list)
  File "/usr/share/backintime/common/encfstools.py", line 519, in list
    for path in _list:
TypeError: 'type' object is not iterable

In conclusion, my backup fails, but I can't even tell which files/directories are causing problems.

I'm using 1.0.34 as provided by ubuntu 14.04.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2014-09-03T16:54:21Z
 * owner: shigi
 * assignee: germar
 * the launchpad url was https://bugs.launchpad.net/bugs/1365072

**Comment by Germar** [comment]
(by germar)
Please use attached patch with 'sudo patch /usr/share/backintime/common/backintime.py < backintime.diff'

This will fix the Type Error in --decode. I'm not yet sure how to filter out the encrypted part in the errorlog. But I'll also fix this soon.

BTW you need to cut of /share/backup/backintime before pasting to --decode

**Comment by Germar** [comment]
(by germar)
Looks like your remote drive is inconsistent. Please run fsck remote.

**Comment by Germar** [comment]
(by shigi)
On 09/03/2014 08:29 PM, Germar wrote:
> Please use attached patch with 'sudo patch
> /usr/share/backintime/common/backintime.py < backintime.diff'

Thanks, works just fine!

> I'm not yet sure how to filter
> out the encrypted part in the errorlog.

Some regexp-magic should do?

**Comment by Germar** [comment]
(by shigi)
On 09/03/2014 08:41 PM, Germar wrote:
> Looks like your remote drive is inconsistent. Please run fsck remote.

I don't think so. Those files are all somewhat special: sockets or
nautilus-actions (see bug #1000543).
They're probably leftovers from an old backup that I manually encrypted
in order to use it as a fresh starting point for incremental backups.
I'll have to delete the files from the current backup and exclude them
in the config. That should fix it.

**Comment by Germar** [comment]
(by germar)
Sure, I've already done a lot of regex foo for the log. But that was always with plaintext mountpoint before encrypted files. Now we have the remote path in plaintext. But that shouldn't be a big problem either.

**Comment by Germar** [comment]
(by germar)
Is '/share/backup/backintime' what you set up in remote path?
I'm afraid you didn't. Or at least you could have set up just 'backintime' for remote path. This makes it very hard to filter out the encrypted part as rsync returns the absolute remote path which BackInTime doesn't know.
